lib/util/transformer/stream-logger.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable promise/prefer-await-to-callbacks */
+import Stream from 'stream';
+import _ from 'lodash';
+import log from '../../logger';
+
+
+const MIN_PRINTABLE_CHAR = 32;
+const MAX_PRINTABLE_CHAR = 126;
+
+class StreamLogger extends Stream.Transform {
+  constructor (direction, verbose = false) {
+    super();
+
+    this._direction = direction;
+    this._verbose = verbose;
+  }
+
+  _transform (data, encoding, callback) {
+    if (this._verbose) {
+      try {
+        this._log(data);
+      } catch (err) {
+        // in case something goes wrong, log it so we can fix the parsing
+        // and also allow everything to continue without interruption
+        log.debug(`Error logging data: ${err.message}`);
+      }
+    }
+    this.push(data);
+    callback();
+  }
+
+  _log (data) {
+    log.debug(`Data ${this._direction === StreamLogger.SEND ? 'sent' : 'received'}, length: ${data.length}`);
+    let i = 0;
+    while (i < data.length) {
```

Comment:
it might be handy to have some explanation about what these nested loops are doing
Also, I have a strange feeling they could perform quite ineffective

Reply:
I'll add some comments.

As for performance, there is no intention of performing particularly well, since (1) this is for debugging purposes, and (2) this is not even close to a performance bottleneck in the Web Inspector system.